source/dub/dependency.d

```diff
@@ -460,11 +486,28 @@ unittest {
 	assert(!a.optional);
 	assert(a.valid);
 	assertThrown(a.version_);
+	assert(a.matches(Version.MASTER));
+	assert(a.matches(Version("1.0.0")));
+	assert(a.matches(Version("0.0.1-pre")));
+	b = Dependency(">=1.0.1");
+	assert(b == a.merge(b));
+	assert(b == b.merge(a));
+	b = Dependency(Version.MASTER);
+	assert(a.merge(m) == m);
+	assert(m.merge(a) == m);
```

Comment:
Oohh, now I see it (misread `b` for `a` in your first comment). `b` by accident happens to be the same as `m` here, but indeed `b` was intended. Will fix.